src/Microsoft.DotNet.Watcher.Tools/Internal/MsBuildFileSetFactory.cs

```diff
@@ -71,20 +73,21 @@ internal MsBuildFileSetFactory(ILogger logger, string projectFile, OutputSink ou
                     var capture = _outputSink.StartCapture();
                     // TODO adding files doesn't currently work. Need to provide a way to detect new files
                     // find files
-                    var exitCode = Command.CreateDotNet("msbuild",
-                        new[]
+                    var processSpec = new ProcessSpec
+                    {
+                        Executable = DotNetMuxer.MuxerPathOrDefault(),
+                        WorkingDirectory = projectDir,
+                        Arguments = new[]
                         {
-                            _projectFile,
+                            "msbuild",
+                             _projectFile,
                             $"/p:_DotNetWatchTargetsLocation={_watchTargetsDir}", // add our dotnet-watch targets
-                            $"/p:_DotNetWatchListFile={watchList}",
-                        }.Concat(BuildFlags))
-                        .CaptureStdErr()
-                        .CaptureStdOut()
-                        .OnErrorLine(l => capture.WriteErrorLine(l))
-                        .OnOutputLine(l => capture.WriteOutputLine(l))
-                        .WorkingDirectory(projectDir)
-                        .Execute()
-                        .ExitCode;
+                            $"/p:_DotNetWatchListFile={watchList}"
+                        }.Concat(BuildFlags),
+                        OutputCapture = capture
+                    };
```

Comment:
I didn't copy this from `Command.CreateDotNet()`. The reason for dropping this is that their Command API is really designed for the CLI's internal usage, and has layers of logic to resolve commands listed in the .csproj. We don't really need that here. We just want to run a process and capture its output.